examples/shader/post_processing.rs

```diff
@@ -162,20 +78,254 @@ fn main_camera_cube_rotator_system(
     }
 }
 
-// Region below declares of the custom material handling post processing effect
+mod post_processing {
+    use bevy::{
+        prelude::*,
+        reflect::TypeUuid,
+        render::{
+            camera::RenderTarget,
+            mesh::Indices,
+            render_resource::{
+                AsBindGroup, Extent3d, PrimitiveTopology, ShaderRef, TextureDescriptor,
+                TextureDimension, TextureFormat, TextureUsages,
+            },
+            texture::BevyDefault,
+            view::RenderLayers,
+        },
+        sprite::{Material2d, Material2dPlugin, MaterialMesh2dBundle},
+        window::{WindowId, WindowResized},
+    };
+
+    pub struct PostProcessingPlugin;
 
-/// Our custom post processing material
-#[derive(AsBindGroup, TypeUuid, Clone)]
-#[uuid = "bc2f08eb-a0fb-43f1-a908-54871ea597d5"]
-struct PostProcessingMaterial {
-    /// In this example, this image will be the result of the main camera.
-    #[texture(0)]
-    #[sampler(1)]
-    source_image: Handle<Image>,
-}
+    impl Plugin for PostProcessingPlugin {
+        fn build(&self, app: &mut App) {
+            app.add_plugin(Material2dPlugin::<PostProcessingMaterial>::default())
+                .add_system(setup_new_post_processing_cameras)
+                .add_system(update_image_to_window_size)
+                .add_system(update_material);
+        }
+    }
+
+    /// To support window resizing, this fits an image to a windows size.
+    #[derive(Component)]
+    struct FitToWindowSize {
+        image: Handle<Image>,
+        material: Handle<PostProcessingMaterial>,
+        window_id: WindowId,
+    }
+    #[derive(Component)]
+    pub struct PostProcessingCamera;
+
+    /// Update image size to fit window
+    fn update_image_to_window_size(
+        windows: Res<Windows>,
+        mut image_events: EventWriter<AssetEvent<Image>>,
+        mut images: ResMut<Assets<Image>>,
+        mut post_processing_materials: ResMut<Assets<PostProcessingMaterial>>,
+        mut resize_events: EventReader<WindowResized>,
+        fit_to_window_size: Query<&FitToWindowSize>,
+    ) {
+        for resize_event in resize_events.iter() {
+            for fit_to_window in fit_to_window_size.iter() {
+                if resize_event.id == fit_to_window.window_id {
+                    let size = {
+                        let window = windows.get(fit_to_window.window_id).expect("PostProcessingCamera is rendering to a window, but this window could not be found");
+                        Extent3d {
+                            width: window.physical_width(),
+                            height: window.physical_height(),
```

Comment:
This is incorrect, same as noted here https://github.com/bevyengine/bevy/pull/5902/files#r1014476079